sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SymbolicExecution/SymbolicValues/SymbolicValue.cs

```diff
@@ -189,47 +203,42 @@ public bool IsNull(ProgramState programState)
         protected IEnumerable<ProgramState> ThrowIfTooMany(IEnumerable<ProgramState> states)
         {
             var stateList = states.ToList();
-            if (stateList.Count >= ExplodedGraph.MaxInternalStateCount)
+            if (stateList.Count >= BaseExplodedGraph.MaxInternalStateCount)
             {
                 throw new TooManyInternalStatesException();
             }
 
             return stateList;
         }
 
-        public virtual IEnumerable<ProgramState> TrySetConstraint(SymbolicValueConstraint constraint, ProgramState currentProgramState)
+        public virtual IEnumerable<ProgramState> TrySetConstraint(SymbolicValueConstraint constraint,

```

Comment:
Agree. I have a strong feeling this can be refactored to look much nicer, but I don't understand it well enough to do it just yet.